iothub/device/src/ConnectionStatusChangeReason.cs

```diff
@@ -8,13 +8,55 @@ namespace Microsoft.Azure.Devices.Client
     /// </summary>
     public enum ConnectionStatusChangeReason
     {
+        /// <summary>
+        /// This signifies that the client is connected, and ready to be used.
+        /// This connection status change reason is returned with a connection status of <see cref="ConnectionStatus.Connected"/>.
+        /// </summary>
         Connection_Ok,
+
+        /// <summary>
+        /// This signifies that the SAS token associated with the client has expired, and cannot be renewed.
+        /// The supplied credentials need to be fixed before a connection can be established.
+        /// NOTE: This status change reason is currently not used in the client library.
+        /// </summary>
         Expired_SAS_Token,
+
+        /// <summary>
+        /// This signifies that the device/ module has been deleted or marked as disabled (on your hub instance).
+        /// Fix the device/ module status in Azure before attempting to create the associated client instance.
+        /// This connection status change reason is returned with a connection status of <see cref="ConnectionStatus.Disconnected"/>
+        /// </summary>
         Device_Disabled,
+
+        /// <summary>
+        /// This signifies that incorrect credentials were supplied to the client instance.
+        /// The supplied credentials need to be fixed before a connection can be established.
+        /// This connection status change reason is returned with a connection status of <see cref="ConnectionStatus.Disconnected"/>
+        /// </summary>
         Bad_Credential,
+
+        /// <summary>
+        /// This signifies that the client was disconnected due to a transient exception, but the retry policy expired before a connection could be re-established.
+        /// This connection status change reason is returned with a connection status of <see cref="ConnectionStatus.Disconnected"/>
+        /// </summary>
         Retry_Expired,
+
+        /// <summary>
+        /// This signifies that the client was disconnected due to loss of network.
+        /// NOTE: This status change reason is currently not used in the client library.
+        /// </summary>
         No_Network,
+
+        /// <summary>
+        /// This connection status change reason when returned with a connection status of <see cref="ConnectionStatus.Disconnected_Retrying"/> signifies that the client is trying to recover from a disconnect due to a transient exception.
+        /// This connection status change reason when returned with a connection status of <see cref="ConnectionStatus.Disconnected"/> signifies that client is disconnected due to a non-retryable exception.
+        /// </summary>
         Communication_Error,
+
+        /// <summary>
+        /// This signifies that the client has been closed gracefully.
+        /// This connection status change reason is returned with a connection status of <see cref="ConnectionStatus.Disabled"/>
+        /// </summary>
```

Comment:
Same here as well. Can you add user action on this status:reason combination nation?

Reply:
sure